app/assets/javascripts/admin/views/user/experience_modal_view.js

```diff
@@ -12,73 +13,108 @@ class ExperienceModalView extends React.PureComponent {
     selectedUserIds: Array<string>,
-    users: Array<APIUserType>,
+    users: Array<APIUserType>
   };
 
   state = {
     domain: null,
-    level: null,
+    level: null
   };
 
   increaseExperience = (): void => {
     this.setExperience(null, true);
   };
 
-  setExperience = (event: ?SyntheticInputEvent, shouldAddValue: boolean = false): void => {
+  setExperience = (
+    event: ?SyntheticInputEvent,
+    shouldAddValue: boolean = false
+  ): void => {
     const { domain, level } = this.state;
-    if (domain && level) {
-      const newUsers = this.props.users.map(user => {
+    if (domain && level !== null) {
+      const newUserPromises = this.props.users.map(user => {
         if (this.props.selectedUserIds.includes(user.id)) {
-          if (shouldAddValue) {
-            if (user.experiences[domain]) {
-              user.experiences[domain] += parseInt(level);
-            } else {
-              user.experiences[domain] = parseInt(level);
-            }
-          } else {
-            user.experiences[domain] = parseInt(level);
+          let newExperienceLevel = parseInt(level);
+
+          if (shouldAddValue && user.experiences[domain]) {
+            newExperienceLevel = user.experiences[domain] + parseInt(level);
           }
 
+          const newUser = update(user, {
+            experiences: { [domain]: { $set: newExperienceLevel } }
+          });
+
           const url = `/api/users/${user.id}`;
           Request.sendJSONReceiveJSON(url, {
-            data: user,
+            data: newUser
           });
-        }
-
-        return user;
-      });
 
-      this.setState({
-        domain: null,
-        level: null,
+          return this.sendUserToServer(newUser, user);
+        }
+        return Promise.resolve(user);
       });
 
-      this.props.onChange(newUsers);
+      this.closeModal(newUserPromises);
     }
   };
 
   deleteExperience = () => {
     if (this.state.domain) {
-      const newUsers = this.props.users.map(user => {
+      const { domain } = this.state.domain;
+      const newUserPromises = this.props.users.map(user => {
         if (this.props.selectedUserIds.includes(user.id)) {
-          delete user.experiences[this.state.domain];
-        }
-        return user;
-      });
+          const newExperiences = _.omit(user.experiences, domain);
+          const newUser = update(user, {
+            experiences: { $set: newExperiences }
+          });
 
-      this.setState({
-        domain: null,
-        level: null,
+          return this.sendUserToServer(newUser, user);
+        }
+        return Promise.resolve(user);
       });
 
-      this.props.onChange(newUsers);
+      this.closeModal(newUserPromises);
     }
   };
 
+  /**
+  * Save a user object to the server using an API call.
+  * @param newUser - A modified user object intended to be saved.
+  * @param oldUser - The original user object of `newUser`. Returned in case API call fails
+  *
+  */
+  sendUserToServer(
```

Comment:
I wonder if we should put this method in an own module. Saving a user (or entities in general) is probably not special to this experience modal. If the code is only here, it's probably going to be re-implemented somewhere else. Admittedly, it's not much code, but with the types and documenting comment it's also "not nothing".
The abstraction could look something like:

```js
type Identifable = {
	id: number | string,
};

function saveObjectToServer<T: Identifable>(
  objectType: string
  newObject: T,
  oldObject: T
): Promise<T> {
  const url = `/api/${objectType}/${newObject.id}`;
  return Request.sendJSONReceiveJSON(url, {
    data: newObject
  }).then(() => Promise.resolve(newObject), () => Promise.reject(oldObject));
}

function saveUserToServer(
  newUser: APIUserType,
  oldUser: APIUserType
): Promise<APIUserType> {
  return saveObjectToServer("users", newUser, oldUser);
}

function saveProjectToServer ...
```

What do you think?